Add emoji search

odmfl opened the pull request:
Added emoji search, from kitten board. Currently it works with English words, because there is a .json file in English only, other .json files should be added for other languages ​​as well it should work. It is based on this library
https://github.com/muan/emojilib

I tried and it works, take a look if you need to make any changes go ahead. I adapted the commits from kittenboard. I think it's a good starting point then others will be able to enhance this function

Fix #35 #197 #527

odmfl committed:
Add emoji search

odmfl committed:
optimize keyboard height,and fix suggestions
problems not visible in many apps like
playstore, kiwi browser...

Fix #660 #745 #747 #275 #418 #507 #345

odmfl committed:
Test apk

odmfl committed:
Fix suggestions again

in the previous commit, some apps still didn't show the hints, now it's totally fixed

ctha42 commented:
Why hasn't this been merged yet?

logical-view commented:
> Hi, i just discover Openboard, the best keyboard i've ever tried on Android, and open-source, great job !

> The only big problem is the suggestions for me, especially on Vivaldi browser : in the URL bar, no suggestions, but right now, while i'm writing here, suggestions work. (As someone mentioned, it doesn't work in Google Play either, but i don't care, it's mainly disabled in the browser)
> 
> When do you think it will be fixed ? Next version ?
> Thanks!

()

When? Commit _739a8dde37d56f961ffdbafa707d674998d8e340_ ?

tkterris commented:
Would it make sense to exclude emojis from auto correct? When I type "Ill" (as the first word in e.g. "I'll see you tomorrow") it gets auto corrected to the emoji "😷". Similarly if I enter "english" (corrected to "🏴󠁧󠁢󠁥󠁮󠁧󠁿") and a couple others.

tkterris commented on app/src/main/java/org/dslul/openboard/inputmethod/latin/Suggest.java:
```diff
@@ -263,6 +266,20 @@ private void getSuggestedWordsForNonBatchInput(final WordComposer wordComposer,
+            List<String> results = EmojiSearch.getInstance().searchExact(typedWordString);
+            if (!results.isEmpty()) {
+                for (int idx = 0; idx < Math.min(2, results.size()); idx++) {
+                    suggestionsContainer.add(idx + 1, new SuggestedWordInfo(results.get(idx),
```
A hacky solution to the issue mentioned in my comment is to insert emojis after the auto-correct entry. So:
```
suggestionsContainer.add(SuggestedWords.INDEX_OF_AUTO_CORRECTION + idx + 1, new SuggestedWordInfo(results.get(idx),
```

logical-view replied:
Add less than three (**1-2**) emoji inside suggestions line, please!

(...correction word) Emoji, correction word, emoji (correction word...) **?**
--->  ` | ` :arrow_lower_left: ` | ` Arrow ` | ` :arrow_lower_right: ` | `

logical-view commented:
Add less than three (**1-2**) emoji inside suggestions line, please!

(...correction word) Emoji, correction word, emoji (correction word...) **?**
--->  ` | ` :arrow_lower_left: ` | ` Arrow ` | ` :arrow_lower_right: ` | `

MattWadbled committed:
Update rows_azerty.xml

Replace rowkey_qwerty by rowkeys_azerty for 600dp width in land mode

Fixes : #668

Helium314 committed:
update dependencies

Helium314 committed:
allow to install debug along release version

Helium314 committed:
Update README.md

Helium314 committed:
allow users to select dictionaries

https://github.com/openboard-team/openboard/pull/569
https://github.com/openboard-team/openboard/pull/578
modified so dictionaries use the correct type instead of always main

Helium314 committed:
add some more dictionaries

see readme for source

Helium314 committed:
allow using all dictionary file versions

removes a hack preventing version < 18, which is really bad for user-added dictionaries

Helium314 committed:
fix dictionary url

Helium314 committed:
fix issues with suggestions

Helium314 committed:
Reduce amount of unwanted automatic space insertions

Helium314 committed:
Exit shift-symbol view on space

Helium314 committed:
add multilingual typing

Helium314 committed:
allow loading external gesture typing library

Helium314 committed:
fix issues with debug version
and some small things

Helium314 committed:
user-adjustable colors

Helium314 committed:
allow removing suggestions

Helium314 committed:
add words to system personal dictionary

Helium314 committed:
fix some theming issues

Helium314 committed:
enable contact suggestions

RHJihan commented:
@odmfl, could you please clone your pull requests here? [https://github.com/Helium314/openboard/](https://github.com/Helium314/openboard/)

odmfl commented:
Guys, I'm very busy, however I created a PR in the fork of @Helium314 
https://github.com/Helium314/openboard
hoping that he manages to continue the project with his fork.
Gesture typing should also be added which works fine here without external closed source libraries, the code could also be revised 
https://github.com/wordmage/openboard

fehnomenal commented:
I'm using the fork right now and gesture typing as well as multi lang typing works like a charm 👍

Helium314 committed:
add experimental dictionaries, and script to create them from sentence lists (wip)

Helium314 committed:
update issue templates

RHJihan committed:
Made Clipboard key default on long press comma

The Clipboard key is set to default on long pressing comma and rearranged for a better user experience.

Helium314 committed:
Merge pull request #7 from RHJihan/new

Made Clipboard key default on long press comma, and re-order other items in that menu

Helium314 committed:
option for navbar to follow theme
fixes #4

Helium314 committed:
update experimental dictionaries, and add experimental emoji dictionary

Helium314 committed:
fix issue with emoji color in more suggestions panel

Helium314 committed:
improve UI for user-added dictionaries

Helium314 committed:
fix color dialog title

Helium314 committed:
copy use-contacts setting to correction settings

Helium314 committed:
update readmes

RHJihan committed:
Added "About" page on 'Settings' (#6)

* Added "About" page on 'Settings'

* configure @string/app_slogan

* auto retrieve version

* Update links in about screen

---------

Co-authored-by: Helium314 <helium314@mailbox.org>

Helium314 committed:
fix issues with keys texts not using user-defined color

Helium314 committed:
fix issues with adding words to personal dictionary

Helium314 committed:
improve issues with key text colors (still happening in more suggestions view)

Helium314 committed:
rename default english dictionary to en_US (because it is), add en_GB dictionary

RHJihan committed:
Fixed dictionary icon (#10)

* Fixed dictionary icon
* fixed about icon dimension

RHJihan committed:
support spell checker for Bengali (Bangladesh) (#11)

Helium314 committed:
Update readme

Helium314 committed:
fix navbar color setting state

Helium314 committed:
add PR template and experimental italian dictionary

Helium314 committed:
update readme

Helium314 committed:
update PR template

BlackyHawky committed:
Update Emojis to version 15

Helium314 committed:
Merge pull request #25 from BlackyHawky/Emoji_v15

Update Emojis to version 15

Helium314 committed:
update experimental emoji dictionary

Helium314 committed:
improve emoji deletion

Helium314 committed:
remove unnecessary code duplication for azerty

was already the case for other layouts and other rows of azerty

Helium314 committed:
set currency hint using currency key style

Helium314 committed:
add armenian, russian and turkish currency keys

and move pound and rupee key styles to separate files, to match others

Helium314 committed:
rename user theme to custom theme

Helium314 committed:
allow setting functional and space key colors separately

not exposed to user
also fixes minor inconsistencies

Helium314 committed:
move getting custom colors to a separate function

and rename some more variables to "custom"
wip: navigation bar color currently broken

Helium314 committed:
fix bad interactions of navbar color and custom theme

Helium314 committed:
missed rename from user to custom

Helium314 committed:
improve coloring of some key icons

Helium314 committed:
insert phantom space after some more characters

Helium314 committed:
properly apply user defined theme immediately

Helium314 committed:
Update README.md

Helium314 committed:
fix minor issues with user colors

BlackyHawky committed:
Fix emoji page indicator background amoled

Helium314 committed:
Merge pull request #26 from BlackyHawky/Fix_emoji_page_indicator_background_amoled

Fix emoji page indicator background amoled

BlackyHawky committed:
Added 11 predefined themes based on material color

BlackyHawky committed:
Fix navbar color with predefined themes

BlackyHawky committed:
Split number row

Helium314 committed:
Merge pull request #27 from BlackyHawky/Split_number_row

Split number row

BlackyHawky committed:
Merge branch 'Helium314:new' into Theme_&_Navbar_color_working

odmfl committed:
Merge pull request #8 from BlackyHawky/Theme_&_Navbar_color_working

Theme & navbar color working

odmfl committed:
Delete android-build.yml

odmfl committed:
Update android.yml

odmfl committed:
Update config.xml

odmfl committed:
Tt

odmfl committed:
Update KeyboardSwitcher.java

RHJihan commented:
@odmfl, I guess you should close this here. All your commits in the master branch are causing a notification to the participants.

logical-Edward commented:
@odmfl rename your fork package to openboard-renewed / openboard-updated ?
> A renewed OpenBoard keyboard, FOSS, updated to integrate changes / improvements that have been [...]
